app/models/grouping.rb

```diff
@@ -728,8 +728,28 @@ def get_next_grouping(current_role, reversed)
     next_grouping
   end
 
+  def get_next_random_unmarked(current_role)
+    groupings = get_all_unmarked(current_role)
+    groupings.order('RANDOM()').first
+  end
+
   private
 
+  def get_all_unmarked(current_role)
```

Comment:
Don't separate this out into a private method, you can put the code into `get_next_random_unmarked`. For testing purposes you can just check that the return value is one of a few different possible values.

(This way you don't have to use `__send__` in a test, which is a bit of a code smell.)